src/Dapr.Actors/Client/ActorProxyFactory.cs

```diff
@@ -42,6 +48,7 @@ public TActorInterface CreateActorProxy<TActorInterface>(ActorId actorId, string
         public ActorProxy Create(ActorId actorId, string actorType, ActorProxyOptions options = null)
         {
             var actorProxy = new ActorProxy();
+            this.daprInteractor.SetDaprApiToken(this.DefaultOptions.DaprApiToken);
```

Comment:
So the problem here is that you're updating a shared instance. I would expect this line of code to create a new `DaprHttpInteractor`. 

Think about what happens when you call Create twice with a different API token. Both proxies would share an `DaprHttpInteractor` instance, and so this line of could would update both of them to use the second API token (last one wins).